### PR TITLE
[Enhancement] Clear storage after lake table been dropped

### PR DESCRIPTION
**Pull request by sduzh**
Modifications:
1. Always put a dropped table into CatalogRecycleBin whether it is a force drop or not.
2. CatalogRecycleBin can save multiple tables with the same name.
3. Does not write edit log for deleting staros's shard anymore when deleting a lake table.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 65 / 104 (62.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableAction.java|0|33|00.00%|[25, 26, 27, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 43, 44, 47, 48, 49, 50, 51, 53, 54, 55, 57, 58, 60, 61, 62, 63, 64, 65, 66]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[135]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[65, 66]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[384]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|46|48|95.83%|[142, 380]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Daemon.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 105 (93.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[135]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[65, 66]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[384]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|46|48|95.83%|[142, 381]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableAction.java|33|33|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Daemon.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/common/util/Daemon.java [hunk 1]
```diff
@@ -97,6 +86,12 @@ public void setInterval(long intervalMs) {
         this.intervalMs = intervalMs;
     }
 
+    public void wakeup() {
+        lock.lock();
+        this.ready.signal();
+        lock.unlock();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/util/Daemon.java:92]:
```suggestion
        lock.lock();
        try {
            this.ready.signal();
        } finally {
            lock.unlock();
        }
```

File: fe/fe-core/src/main/java/com/starrocks/common/util/Daemon.java [hunk 2]
```diff
@@ -113,15 +108,12 @@ public void run() {
         while (!isStop.get()) {
             try {
                 runOneCycle();
+                lock.lock();
+                ready.await(intervalMs, TimeUnit.MILLISECONDS); // Ignore the result intentionally
+                lock.unlock();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/util/Daemon.java:113]:
```suggestion
                lock.lock();
                try {
                    ready.await(intervalMs, TimeUnit.MILLISECONDS); // Ignore the result intentionally
                } finally {
                    lock.unlock();
                }
```

**Comment by sduzh** [comment]
run starrocks_fe_unittest

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 105 (93.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[135]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[72, 73]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[384]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|46|48|95.83%|[142, 381]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableAction.java|33|33|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Daemon.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java [hunk 3]
```diff
@@ -521,6 +521,10 @@ public synchronized boolean recoverTable(Database db, String tableName) {
         }
 
         Table table = recycleTableInfo.getTable();
+        if (idToRecycleTime.get(table.getId()) <= 0) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java:524]: If delete action is executed failed and FE is restarted, recycle time will be restored to the current time.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 98 / 105 (93.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[126]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[72, 73]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[384]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|46|48|95.83%|[142, 381]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableAction.java|33|33|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Daemon.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|

**Review by Astralidea — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/Daemon.java (hunk 2 above)
Comment (Astralidea) [reply fe/fe-core/src/main/java/com/starrocks/common/util/Daemon.java:113]: +1

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 117 / 126 (92.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[126]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[72, 73]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[384]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|65|69|94.20%|[111, 142, 175, 380]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableAction.java|33|33|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/Daemon.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 119 / 129 (92.25%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|0|1|00.00%|[126]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|1|00.00%|[694]|
|:large_blue_circle:|com/starrocks/rpc/LakeServiceClient.java|0|2|00.00%|[72, 73]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[384]|
|:large_blue_circle:|com/starrocks/common/util/Daemon.java|12|13|92.31%|[102]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|65|69|94.20%|[111, 142, 175, 383]|
|:large_blue_circle:|com/starrocks/lake/DeleteLakeTableAction.java|33|33|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|2|2|100.00%|[]|

**Pull request closed without merging**